code/obj/flock/structure/gnesisturret.dm

```diff
@@ -120,25 +118,26 @@
 					logTheThing(LOG_COMBAT, src, "Flock gnesis turret at [log_loc(src)] belonging to flock [src.flock?.name] fires at [constructTarget(src.target)].")
 					SPAWN(0)
 						for(var/i in 1 to src.current_projectile.shot_number) //loop animation until finished
-							muzzle_flash_any(src, 0, "muzzle_flash")
+							muzzle_flash_any(src, get_angle(src, src.target), "muzzle_flash")
 							sleep(src.current_projectile.shot_delay)
 					shoot_projectile_ST_pixel_spread(src, current_projectile, target, 0, 0 , spread)
 
 	proc/seek_target()
 		var/list/target_list = list()
-		for (var/mob/living/C in range(src.range,src.loc))
+		for (var/mob/living/C in hearers(src.range,src.loc))
```

Comment:
hearers?

Reply:
hearers is line-of-sight, it's very confusingly named. It doesn't account for not being able to shoot through windows, but it should stop trying to shoot through walls.